### DISCUSSION TITLE
Smooth recovery from invalid configs

### DISCUSSION THREAD

**Discussion by nik-rev** [helix-editor/helix#12342] [Ideas]
If you have an invalid config key, your entire config will be disabled

But this can be really annoying for people who have custom PRs merged into their personal forks. I currently have https://github.com/helix-editor/helix/pull/11285 merged and have the following config set:

```toml
[keys.normal.space]
E = "file_browser"
```

But since this is only valid in that PR and not on master, everytime I try to use a different branch that doesn't have my config my entire setup stops working:

![image](https://github.com/user-attachments/assets/867eb01f-f275-46f7-a4af-ad120e34d82d)

So what I usually do is comment out that part of config every time. Since I have quite a lot of branches for Helix, and test various things out time to time, this ends up being a common procedure

Imo, it would be better for user experience if you still got the `Bad config` warning and forced to press enter, but at least your *other* settings should continue to work.

**Comment by the-mikedavis** [comment]
This is probably not a straightforward change to make because validations like checking command names are done at parsing time. This is something that could be resolved in the future as we switch config to drop serde and use scheme: https://github.com/helix-editor/helix/issues/8853

**Reply by gabydd** [reply]
@NikitaRevenco the issue for this is actually https://github.com/helix-editor/helix/issues/10389

**Reply by RoloEdits** [reply]
This actually brings up a great point for custom commands. They would need to be parsed first in order for the keymaps to be able to use them.

(added a todo)

**Reply by nik-rev** [reply]
> @NikitaRevenco the issue for this is actually #10389

Thanks. i posted a comment there
